### PR TITLE
feat: v4 menu docs & tests

### PR DESCRIPTION
**Pull request by bgutsol**
# Purpose of PR

Documentation, tests coverage and some minor fixes for `Menu`  component

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

**Review by suevalov — APPROVED** [review]
Looks great!

File: packages/components/menu/Menu.mdx [hunk 1]
```diff
@@ -8,4 +8,233 @@ storybook: 'https://f36-storybook.contentful.com/?path=/story/components-menu--d
...
+```
+
+## Content recommendations
+
+- Use an interactive element such as `button` for `Menu.Trigger`
+
+## Accessibility
+
+- When the menu is opened, focus is moved to the first menu item. If you set `autoFocus`to `false`, it will not move the focus.
+- Whne the menu is closed, focus returned back to the trigger element.
```
Comment (burakukula) [inline packages/components/menu/Menu.mdx:232]:
```suggestion
- When the menu is closed, focus returned back to the trigger element.
```

File: packages/components/menu/test/Menu.test.tsx [hunk 2]
```diff
@@ -1,11 +1,239 @@
...
+
+    await waitFor(() => {
+      expect(getByRole('menu')).toBeInTheDocument();
+    });
+
+    userEvent.click(getByTestId('itemToClick'));
+    expect(handleClose).not.toHaveBeenCalled();
+  });
+
+  it('should focus FIRST item when menu open', async () => {
```
Comment (burakukula) [inline packages/components/menu/test/Menu.test.tsx:128]:
```suggestion
  it('should focus FIRST item when menu is open', async () => {
```

**Review by burakukula — APPROVED** [review]
Nicely done! 🏄

**Pull request merged**